### PR TITLE
Allow the user to customise the dmesg whitelist.

### LINKED ISSUE #213: Make dmesg filter rules configurable.
As it stands the rules are hard-coded in `platform.py`, the user should be able to configure these in the config file and on a per-platform basis.

### PR DESCRIPTION
**Pull request by vext01**
This changes allows the user to customise the dmesg whitelist.

In the end I *did* opt for the callback.

Queries:
 * Should the `CUSTOM_DMESG_WHITELIST` config func be lowercase or remain all caps?
 * I'm thinking of renaming `get_allowed_dmesg_lines()` to `get_dmesg_whitelist()`. Thoughts?

Comments? OK?

Fixes #213 

I have no idea what that cache file is. Going to kill it.

### CODE REVIEW

**Comment by ltratt** [comment]
W.r.t. to the queries, in order: lowercase; yes.

**Comment by vext01** [comment]
OK. I'd also like to (manually) test this on a dummy benchmark with puts lines in the dmesg. If such a thing is possible...

**Comment by ltratt** [comment]
I don't see how that's possible, to be honest..

**Comment by vext01** [comment]
I've tested this on Linux using a hack I found on SO.

By invoking
```
echo blah > /dev/kmsg
```

on Linux during a benchmark, I can simulate new dmesg lines and write custom whitelists to match them.

All seems well.

Also added docs.

Any comments?

**Comment by ltratt** [comment]
Would it be easier to squash this before I start the review?

**Comment by vext01** [comment]
Can do.

**Comment by ltratt** [comment]
OK, let's do it.

**Comment by vext01** [comment]
splat

File: README.md [hunk 1]
```diff
@@ -425,6 +425,38 @@ PRE_EXECUTION_CMDS = [
...
+is a collection of regex patterns which are used to decide if a line in the
+dmesg buffer is a cause for concern. If a new line appears in the dmesg during
+benchmarking, and the line is not matched by at least one whitelist pattern,
+then Krun will flag the process execution as ``errored'' and email the user.
+
+From time to time you may find that you need to customise the whitelist. This
+is achieved by adding a callback named `custom_dmesg_whitelist` into your
+config file. The callback is passed the default list of patterns for your
+platform and must return a new list of patterns. In the implementation of your
+callback you have the choice to base your custom whitelist on the defaults or
```
Comment (ltratt) [inline README.md]: "or to define"

**Comment by ltratt** [comment]
Apart from that one typo, this looks good to me.

**Comment by vext01** [comment]
also s/lines/patterns/ in the last commit.

Squash?

**Comment by ltratt** [comment]
Yep, let's squash.

**Comment by vext01** [comment]
splat.

**Pull request merged**